notes/lecture7.tex

```diff
@@ -0,0 +1,137 @@
+\end{equation*}
+This implies
+\begin{equation*}
+\eta T (f(\bar{x}) - f(x^*)) \leq \sum_{k=0}^T \eta \langle\nabla f(x_{k+1}), x_{k+1} - x^*\rangle\\
+\leq \eta^2 \beta d + R^2,
+\end{equation*}
+which can be rewritten as
+\begin{equation*}
+f(\bar{x}) - f(x^*) \leq \frac{\eta \beta d}{T} + \frac{R^2}{\eta T}.
+\end{equation*}
+Setting $\eta = \frac{R}{\sqrt{\beta d}}$, this gives us
+\begin{equation*}
+f(\bar{x}) - f(x) \leq \frac{2 \sqrt{\beta d}}{T} R.
+\end{equation*}
+Therefore, for $T \geq 4 \sqrt{\frac{\beta}{d}} R$, we have
+\begin{equation*}
+f(\bar{x}) - f(x) \leq \frac{d}{2}.
+\end{equation*}
+\end{proof}
+
+This lemma was initially proven by Allen-Zhu and Orecchia in 2014 in their article \textit{Linear Coupling: An Ultimate Unification of Gradient and Mirror Descent}. From the lemma, our analysis of the runtime of the algorithm naturally follows.
+
+\begin{proof}
+1) If a function $f$ is $\beta$-smooth and convex (not strongly convex), the Lemma gives that after $T(d)=4R\sqrt{\frac{\beta}{d}}$ steps, 
+\[f(\bar{x})- f(x^*)\leq \frac{d}{2}\qquad\text{, where } \bar{x} = \frac{1}{T}\sum_{\tau=0}^{T-1}x_k.\]
+
+This means after $T(d)$ steps, Nesterov's accelerated gradient descent method reduces the function value error by half. For the complete Nesterov's method, after every $T(d)$ steps, let $\bar{x}$ be the new initial state and restart the updating algorithm from the beginning. %Then the error will keep decreasing until it is less then $\epsilon$.\\
+
+\indent By applying the Lemma $n$ times, we have the total iterations before $f(\bar{x}_{(n)})-f(x^*)\leq\frac{d}{2^{n}}\leq\epsilon$,
+\begin{align*}
+t(\epsilon) &= \cO \left(4R\sqrt{\frac{\beta}{d/2^{n-1}}} + 4R\sqrt{\frac{\beta}{d/2}} + 4R\sqrt{\frac{\beta}{d}}\right)\\
+  &= \cO \left(4R\sqrt{\frac{\beta}{d/2^{n-1}}}\right)\\
+  &= \cO \left(\sqrt{\frac{\beta}{\epsilon}}\right).
+\end{align*}
+\indent This can also be written as: 
+\[\epsilon(t) = \cO \left(\frac{\beta}{t^2} \right)\]
+
+2) If function $f$ is $\beta$-smooth and $\alpha$-strongly convex, the Lemma gives that after constant $T=4\sqrt{\frac{2\beta}{\alpha}}$ steps,
+\[\|\bar{x} - x^*\|^2 \leq \frac{1}{2}\|x_0 - x^*\|^2 \qquad\text{, where } \bar{x} = \frac{1}{T}\sum_{\tau=0}^{T-1}x_k.\]
+
+\indent Using the same trick in 1), we apply the Lemma n times before $\|\bar{x}_{(n)} - x^*\|^2 \leq \frac{1}{2^n}\|x_0 - x^*\|^2 \leq \epsilon$. Then the total iterations are:
+\begin{align*}
+t(\epsilon) &= \cO(nT)\\
+            &= \cO \left (4\sqrt{\frac{2\beta}{\alpha}}\log_2 \left (\frac{\|x_0 - x^*\|^2}{\epsilon} \right) \right)\\
+            &= \cO \left(-\sqrt{\frac{\beta}{\alpha}}\log_2\epsilon \right).
+\end{align*}
+
+\indent In this case, Nesterov's accelerated gradient descent method yields linear convergence:
+\[\epsilon(t) = \cO \left (2^{-t\sqrt{\frac{\alpha}{\beta}}} \right).\]
```

Comment:
When you solve for epsilon the constant factor goes in the exponent and on in the base (which is not equivalent). So you'd have to write `e^{-\Omega(...)}` if you want to use big-Oh notation.

Reply:
Is it correct to write in big-Omega notation? I'm not sure I completely follow.

Reply:
Yes, it's a bit confusing because of the negative sign in the exponent, which implies that big Oh has to become big Omega. Note that an upper bound of 2^{- O(f)} is trivial, because 0 is always in O(f), and 2^{-0}=1. That's why you need to write 2^{-\Omega(f)}. If you find this to confusing, please just write it as 2^{-c f}, for some constant c>0. Thanks.